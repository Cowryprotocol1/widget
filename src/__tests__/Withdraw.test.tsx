import { fireEvent, render } from "@testing-library/react";
import WithdrawModal from "../withdrawal";
import React from 'react';
const UserWalletBalance: string = "3000";

describe("Withdrawal Component", () => {
  // defining the component
  const testComponent =()=> render(<WithdrawModal UserWalletBalance={UserWalletBalance}/>)

  it(" ❌ renders Withdrawal Component correctly'", () => {
   const {asFragment} = testComponent();
   expect(asFragment()).toMatchSnapshot();
  });

  it(" ❌ count number of input in the form'", async () => {
    let component = testComponent();
    const formInputs = component.getAllByTestId('form')
    expect(await formInputs).toHaveLength(6);
   });

  it(" ❌ Submit Form Data'", () => {
    const {getByTestId, getByText, getByPlaceholderText} = testComponent();

    const amount:any = getByPlaceholderText('Withdraw Amount')
    fireEvent.change(amount,{target: {value: '1000'}})
    expect(amount.value).toBe('1000');

    const bank:any = getByPlaceholderText('Bank Name')
    fireEvent.change(bank,{target: {value: 'GTB'}})
    expect(bank.value).toBe('GTB');

    const desc:any = getByPlaceholderText('Transaction Description')
    fireEvent.change(desc,{target: {value: 'Test Desc'}})
    expect(desc.value).toBe('Test Desc');

    const accNum:any = getByPlaceholderText('Account Number')
    fireEvent.change(accNum,{target: {value: '0987654321'}})
    expect(accNum.value).toBe('0987654321');

    const accName:any = getByPlaceholderText('Account Name')
    fireEvent.change(accName,{target: {value: 'Daniel Olagunju'}})
    expect(accName.value).toBe('Daniel Olagunju');

    const phone:any = getByPlaceholderText('Phone Number');
    fireEvent.change(phone,{target: {value: '08012345678'}});
    expect(phone.value).toBe('08012345678');

    const button = getByTestId('submit_btn')
    fireEvent.click(button);
    expect(getByText('Confirm Withdrawal')).toBeInTheDocument();
  });
});