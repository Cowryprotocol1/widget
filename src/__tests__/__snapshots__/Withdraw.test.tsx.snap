// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Withdrawal Component  ❌ renders Withdrawal Component correctly' 1`] = `
<DocumentFragment>
  <div
    class="bg-white w-[35%] py-4 px-[2.5%] rounded"
  >
    <div
      class="w-full p-3 flex flex-row bg-[#062638] rounded-lg"
    >
      <div
        class="flex text-sm flex-col justify-start"
      >
        <p
          class="text-white font-thin mb-2 text-left"
        >
          Request Withdraw
        </p>
        <p
          class="text-white font-thin text-left md:text-xs text-[0.7rem]"
        >
          You can withdraw from your cowry account by paying into the wallet details given below.
        </p>
      </div>
      <img
        alt="logo_name_c"
        class="w-[65px] md:w-[90px]"
        src="deposit.png"
      />
    </div>
    <div
      class="flex flex-col items-center gap-6 mt-8 w-[100%]"
    >
      <div
        class="w-[100%] relative mb-2"
      >
        <input
          class="px-4 border-1 h-[45px] bg-[#EDEDED] text-black w-full md:w-[100%] text-xs  font-thin rounded"
          data-testid="form"
          maxlength="100"
          name="amount"
          placeholder="Withdraw Amount"
          type="text"
          value=""
        />
        <p
          class="text-[9px] text-[#414141] absolute top-[-0.5rem] md:top-[-0.9rem] px-1 left-0 md:left-0 bg-transparent"
        >
          Withdraw Amount
        </p>
        <p
          class="text-[9px] text-[#414141] px-1 text-right mb-[-1rem] bg-white"
        >
          Wallet balance:
          <span
            class="text-brand_primary_green"
          >
            NGN 3,000.00
          </span>
        </p>
      </div>
      <div
        class="w-[100%] relative mb-2"
      >
        <input
          class="px-4 border-1 h-[45px] bg-[#EDEDED] text-black w-full md:w-[100%] text-xs  font-thin rounded"
          data-testid="form"
          maxlength="100"
          name="bank"
          placeholder="Bank Name"
          type="text"
          value=""
        />
        <p
          class="text-[9px] text-[#414141] absolute top-[-0.5rem] md:top-[-0.9rem] px-1 left-0 md:left-0 bg-transparent"
        >
          Bank Name
        </p>
      </div>
      <div
        class="w-[100%] relative mb-2"
      >
        <input
          class="px-4 border-1 h-[45px] bg-[#EDEDED] text-black w-full md:w-[100%] text-xs  font-thin rounded"
          data-testid="form"
          maxlength="100"
          name="description"
          placeholder="Transaction Description"
          type="text"
          value=""
        />
        <p
          class="text-[9px] text-[#414141] absolute top-[-0.5rem] md:top-[-0.9rem] px-1 left-0 md:left-0 bg-transparent"
        >
          Transaction Description
        </p>
      </div>
      <div
        class="w-[100%] relative mb-2"
      >
        <input
          class="px-4 border-1 h-[45px] bg-[#EDEDED] text-black w-full md:w-[100%] text-xs  font-thin rounded"
          data-testid="form"
          maxlength="10"
          name="account_number"
          placeholder="Account Number"
          type="text"
          value=""
        />
        <p
          class="text-[9px] text-[#414141] absolute top-[-0.5rem] md:top-[-0.9rem] px-1 left-0 md:left-0 bg-transparent"
        >
          Account Number
        </p>
      </div>
      <div
        class="w-[100%] relative mb-2"
      >
        <input
          class="px-4 border-1 h-[45px] bg-[#EDEDED] text-black w-full md:w-[100%] text-xs  font-thin rounded"
          data-testid="form"
          maxlength="100"
          name="account_name"
          placeholder="Account Name"
          type="text"
          value=""
        />
        <p
          class="text-[9px] text-[#414141] absolute top-[-0.5rem] md:top-[-0.9rem] px-1 left-0 md:left-0 bg-transparent"
        >
          Account Name
        </p>
      </div>
      <div
        class="w-[100%] relative mb-2"
      >
        <input
          class="px-4 border-1 h-[45px] bg-[#EDEDED] text-black w-full md:w-[100%] text-xs  font-thin rounded"
          data-testid="form"
          maxlength="100"
          name="phone"
          placeholder="Phone Number"
          type="text"
          value=""
        />
        <p
          class="text-[9px] text-[#414141] absolute top-[-0.5rem] md:top-[-0.9rem] px-1 left-0 md:left-0 bg-transparent"
        >
          Phone Number
        </p>
      </div>
      <button
        class="bg-gray-300 text-black rounded px-4 py-2 text-xs"
        data-testid="submit_btn"
        disabled=""
      >
        Withdraw
      </button>
    </div>
  </div>
</DocumentFragment>
`;
